lib/api/controllers/securityController.js

```diff
@@ -329,6 +330,9 @@ module.exports = function SecurityController (kuzzle) {
         if (newRequestObject.data._id !== undefined) {
           pojoUser._id = newRequestObject.data._id;
         }
+        else {
+          pojoUser._id = uuid.v4();
```

Comment:
This was actually the situation this PR tries to solve.

The createUser action should create a real user when no id is provided and not update the anonymous user.